### ISSUE TITLE
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff

### ISSUE THREAD

**Issue by mefikru** [mesonbuild/meson#9158]
**Describe the bug**
Just on run meson
```
Traceback (most recent call last):
  File "c:\users\mefik\appdata\local\programs\python\python39\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\mefik\appdata\local\programs\python\python39\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\mefik\AppData\Local\Programs\Python\Python39\Scripts\meson.exe\__main__.py", line 7, in <module>
  File "c:\users\mefik\appdata\local\programs\python\python39\lib\site-packages\mesonbuild\mesonmain.py", line 318, in main
    setup_vsenv()
  File "c:\users\mefik\appdata\local\programs\python\python39\lib\site-packages\mesonbuild\mesonmain.py", line 89, in setup_vsenv
    bat_info = json.loads(bat_json)
  File "c:\users\mefik\appdata\local\programs\python\python39\lib\json\__init__.py", line 341, in loads
    s = s.decode(detect_encoding(s), 'surrogatepass')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 639: invalid start byte
```

**To Reproduce**
vswhere return json file with 0xff in Display name.
file attached - https://drive.google.com/file/d/1AJEdApiiJb0x-IgBPE8L2gUGYc-ZVfyL/view?usp=sharing

**system parameters**
Windows 10 VL LTSC 1809 x64 ru
Visual Studio Community 2019 16.11.1 ru
Python 3.9.6

**Comment by eli-schwartz** [comment]
```
$ chardetect /tmp/1.json 
/tmp/1.json: IBM866 with confidence 0.9290660824351996
```

It's bizarre that Visual Studio Community 2019 is ignoring the -utf8 command line switch which meson specifically uses, and emitting cp866 encoded data instead.
https://devblogs.microsoft.com/setup/vswhere-now-supports-utf8-to-override-console-code-page/

This should in theory be impossible, since it should error out on invalid options. I'm not sure what we can do about this though, since it would seem your vswhere is broken.

**Comment by mefikru** [comment]
> It's bizarre that Visual Studio Community 2019 is ignoring the -utf8 command line switch which meson specifically uses, and emitting cp866 encoded data instead.
Nope, there is no -utf8 switch used. 
mesonmain.py, line 77:
`    
bat_json = subprocess.check_output(
        [
            str(bat_locator_bin),
            '-latest',
            '-prerelease',
            '-requiresAny',
            '-requires', 'Microsoft.VisualStudio.Component.VC.Tools.x86.x64',
            '-products', '*',
            '-format',
            'json',
        ]
    )
`

So, I add -utf8 switch and it solve the problem.

**Comment by eli-schwartz** [comment]
Ahhh! Sorry. I based my analysis on reading my git checkout of meson. The current meson release is genuinely broken, but it was already fixed in commit a79980da6ba50ef9eaea396528c9c8a395c4997d which is why I saw it using the -utf8 switch.

Regrettably, the PR implementing it, #9054, got overlooked and did not get tagged as a bugfix for meson 0.59.1, however, I've just added it to the 0.59.2 milestone.
